libtransmission/peer-mse.cc

```diff
@@ -78,18 +78,18 @@ namespace tr_message_stream_encryption
     return tr_rand_obj<DH::private_key_bigend_t>();
 }
 
-[[nodiscard]] auto generatePublicKey(DH::private_key_bigend_t const& private_key) noexcept
-{
-    auto const private_key_wi = wi::import_bits<wi::private_key_t>(private_key);
-    auto const public_key_wi = math::wide_integer::powm(wi::generator, private_key_wi, wi::prime);
-    return wi::export_bits(public_key_wi);
-}
-
 DH::key_bigend_t DH::publicKey() noexcept
 {
+    auto generate_public_key = [](DH::private_key_bigend_t const& private_key) noexcept
+    {
+        auto const private_key_wi = wi::import_bits<wi::private_key_t>(private_key);
+        auto const public_key_wi = math::wide_integer::powm(wi::generator, private_key_wi, wi::prime);
+        return wi::export_bits(public_key_wi);
+    };
```

Comment:
This lambda is only used once, maybe we should just inline it.

Reply:
good call. inlined in 33f9a54f897d57fe094843fb4e96a942366efa44